museek/plugin/sanity_check_observation_plugin.py

```diff
@@ -220,3 +222,56 @@ def create_plots_of_scan_data(self, data: TimeOrderedData):
         plt.hist(data.elevation.squeeze.flatten(), bins=200)
         plt.xlabel('elevation')
         self.savefig(description='Elevation histogram of all dishes during scan.')
+
+
+    def check_closeness_to_sunrise_sunset(self, data: TimeOrderedData, report_writer: ReportWriter):
+
+        """
+        Check if observation was done within 30 min of sunset or sunrise.
+        :param data: the `TimeOrderedData` to check
+        :param report_writer: the `ReportWriter` object to handle the report
+        """
+
+        import ephem
+        from datetime import datetime, timedelta
+        import numpy as np
+
+        observer = ephem.Observer()
+        observer.lat = '-30.7130'  # Latitude of Meerkat
+        observer.lon = '21.4203'  # Longitude of Meerkat
+
+        # Set the observer's date
+        date_time = datetime.utcfromtimestamp(float(data.original_timestamps[0]))
+        observer.date = date_time
+
+        # Calculate sunset and sunrise times
+        sunset_time = observer.previous_setting(ephem.Sun())  # Sunset
```

Comment:
What if the observation starts with the sun still up? It's an extreme case, but then both `previous_setting` and `next_rising` would be far away... Would this check still work correctly?

Reply:
In very extreme case, the previous_setting() or next_rising() would give the sunset or sunrise date in last day or next day, so I only use the hour, minute and second information to calculate the time difference

Reply:
I guess MeerKAT is close enough to the equator that sunset/rise times don't vary more than a few minutes from day to day? Maybe add a comment that this is an assumption made here. By the way, I think it's actually not out of the ordinary for them to start observing before the sun has fully set.

Reply:
The sunset/rise times vary  a few seconds from day to day

Reply:
That will depend on the time of the year...

Reply:
It is better to add a comment to help people read your code. You are making an assumption which is not obvious immediately

Reply:
30 minutes is a handwaving setting anyway, so a minute isn't going to matter. But I agree. Leave a comment

Reply:
Comments added.